### ISSUE TITLE
Docs build produces 1000+ warnings due to malformed docstrings and autodoc import failures

### ISSUE THREAD

**Issue by utkarshsinha-dev** [deepchem/deepchem#4740]
## 🐛 Bug

Building the documentation locally completes, but produces ~1162 warnings/errors related to malformed docstrings, duplicate rst targets, and autodoc import failures.

While HTML output is generated, the number of warnings suggests many API docs are currently broken or improperly formatted.

## To Reproduce

1.```bash
2.cd docs
3.sphinx-build -b html source build

Example errors from sphinx-build:

C:\Users\HP\deepchem\deepchem\models\torch_models\fno.py
ERROR: Unexpected indentation.

C:\Users\HP\deepchem\deepchem\models\graph_models.py
ERROR: Duplicate target name, cannot be used as a unique reference: "1".

WARNING: autodoc: failed to import class 'DCLightningModule' from module 'deepchem.models'

## Expected behavior

Documentation should build with minimal warnings. Optional dependencies should not break autodoc, and docstrings/rst files should not raise structural errors.

## Environment

* OS: Windows 11
* Python version: 3.10 (conda) 
* DeepChem version: local clone (current main) 
* RDKit version (optional):
* TensorFlow version (optional):
* PyTorch version (optional):
* Any other relevant information:

## Additional context

I’m happy to help fix a subset of these issues (for example docstring formatting or rst label cleanup) if guidance on scope is provided.

**Comment by utkarshsinha-dev** [comment]
Hi, I’d like to work on fixing a subset of these documentation warnings (starting with docstring indentation / duplicate rst targets).

Please let me know if that’s okay, or if there’s a preferred scope to start with. Thanks!

**Comment by ArnabTechiee** [comment]
Hi @rbharath and @utkarshsinha-dev, this is a massive list of warnings! I'd like to help by taking a specific slice of this issue.

Since my recent PRs (#4759 and #4760) have been focused on the NLP and PyTorch stack, I’ll take responsibility for fixing the docstring indentation and autodoc failures in torch_models and feat/smiles_tokenizer. This will ensure that the documentation for the NLP pipeline is clean and professional for the upcoming GSoC cycle. May I proceed with this scope?

**Comment by utkarshsinha-dev** [comment]
Hi @ArnabTechiee 
Sounds good — thanks for volunteering to handle this slice.
I’m not working on those modules currently.
Appreciate you taking this on!
Looking forward to your PR

**Comment by Vritika22Mandapaka** [comment]
Hi, I’d like to work on a small subset of this issue as my first contribution.

I plan to start by fixing one malformed docstring / indentation warning in `deepchem/models/torch_models/fno.py`, then verify the docs build again locally.

Please let me know if that scope sounds good.
